src/librbd/io/ImageRequest.cc

```diff
@@ -779,7 +749,7 @@ uint64_t ImageCompareAndWriteRequest<I>::append_journal_event(
                                                           extent.second,
                                                           m_cmp_bl,
                                                           m_bl,
-                                                          synchronous);
+                                                          false);
```

Comment:
I actually initially intended to drop `flush_entry` as well but discovered that one test passes `true` for that parameter: https://github.com/ceph/ceph/blob/b393ea035a8e8614cccfc031b6a51bb11b9ef1ca/src/test/librbd/journal/test_Replay.cc#L56-L57

I don't know this area of the code well enough to know whether there's an easy way to avoid that in the code (e.g. is there a separate flush call that could be made after appending the event?).

Reply:
> is there a separate flush call that could be made after appending the event?

There is `Journal::flush_event()`.  Its semantics aren't _exactly_ the same but hopefully close enough for injecting an event into the journal in a test.  @trociny Can `wait_event()` on the next line be replaced with `flush_event()` instead of triggering the flush from within `append_io_events()`?

Reply:
> Can `wait_event()` on the next line be replaced with `flush_event()` instead of triggering the flush from within `append_io_events()`?

I'm merging this as is since it's an important bug fix.  @trociny If the answer to the above is true, let's file a cleanup ticket.